engine.go

```diff
@@ -12,18 +12,35 @@ type (
 	// HostCallHandler is a function to invoke to handle when a guest is performing a host call.
 	HostCallHandler func(ctx context.Context, binding, namespace, operation string, payload []byte) ([]byte, error)
 
+	// EngineOption is a struct that holds the various otions that are used during Engine.New
+	EngineOption struct {
+		Ctx    context.Context
+		Host   HostCallHandler
+		Guest  []byte
+		Config *ModuleConfig
+	}
+
+	// Engine is the WASM engine abstraction for WAPC
+	// All engines must implement this interface
 	Engine interface {
 		// Name of the engine. Ex. "wazero"
 		Name() string
+		Options() *EngineOption
 
 		// New compiles a new WebAssembly module representing the guest, and
 		// configures the host functions it uses.
 		//   - host: implements host module functions called by the guest
 		//	 - guest: the guest WebAssembly binary (%.wasm) to compile
 		//   - config: configures the host and guest.
-		New(ctx context.Context, host HostCallHandler, guest []byte, config *ModuleConfig) (Module, error)
+		New(ctx context.Context, host HostCallHandler, guest []byte, config *ModuleConfig) (mod Module, err error)
+
+		// NewWith an option type pattern for New
+		NewWith(engineOpt ...EngineOptionFn) (Module, error)
```

Comment:
ctx and guest are not options. and also returning options is not a usual pattern.